src/sync/installation.ts

```diff
@@ -110,16 +110,30 @@ export const updateTaskStatusAndContinue = async (
 			}
 		}
 
-		statsd.histogram(metricTaskStatus.complete, duration, [`type:${task.task}`, `gitHubProduct:${gitHubProduct}`]);
+		sendTaskMetrics(metricTaskStatus.complete, task, gitHubProduct);
 	} else {
 		updateRepoSyncFields[getCursorKey(task.task)] = edges![edges!.length - 1].cursor;
-		statsd.histogram(metricTaskStatus.pending, duration, [`type:${task.task}`, `gitHubProduct:${gitHubProduct}`]);
+		sendTaskMetrics(metricTaskStatus.pending, task, gitHubProduct);
 	}
 
 	await updateRepo(subscription, task.repositoryId, updateRepoSyncFields);
 	await sendBackfillMessage(data, 0, logger);
 };
 
+const sendTaskMetrics = (metrics: string, task: Task, gitHubProduct: "server" | "cloud", failedCode?: string) => {
+	const duration = task.startTime ? Date.now() - task.startTime : undefined;
+	const tags = {
+		type: task.task,
+		gitHubProduct,
+		failedCode: failedCode || ""
+	};
+	if (duration === undefined) {
+		statsd.increment(metrics, tags);
```

Comment:
The `histogram` doesn't support an undefined number in its signature, so for undefined value (which isn't likely to happen according to our app flow, but from typescript point of view I can't exclude it. Also don't want to use `!` to force convert it), we will only have the `count` metrics for it.